plugins/modules/datadog_monitor.py

```diff
@@ -131,11 +131,6 @@
           - Only available for service checks and metric alerts.
           - Because each of them can have multiple thresholds, we do not define them directly in the query.
           - "If not specified, it defaults to: V({'ok': 1, 'critical': 1, 'warning': 1})."
-    locked:
-        description:
-          - Whether changes to this monitor should be restricted to the creator or admins.
-        type: bool
-        default: false
```

Comment:
Please note that simply removing a module option is a breaking change and cannot be merged without a deprecation period.

Also it seems that the services can apparently be hosted on-premises, in which case this could work a lot longer than for the cloud-hosted solution.

Reply:
Thanks for the feedback. Do you have any reference to that? To my knowledge datadog.com is a cloud hosted monitoring solution with no on-premise equivalent. You can run on-premise agents on your machine, that report metrics to the datadog cloud. There you deploy monitors, which alert on the metrics or logs.

Reply:
> Thanks for the feedback. Do you have any reference to that? To my knowledge datadog.com is a cloud hosted monitoring solution with no on-premise equivalent.

I have no real reference, I found https://www.datadoghq.com/solutions/on-premises-monitoring/ when searching for 'datadog on premises', but that could well be just about the on-premise agents; but then there's also a module parameter called `api_host`.

Reply:
The api_host parameter refers to the different [Datadog API endpoints](https://docs.datadoghq.com/getting_started/site/). The API endpoint changes, depending which site of Datadog the user is using. For example the endpoint for site US is  https://api.datadoghq.com/api/v1/hosts and for site US3 is https://api.us3.datadoghq.com/api/v1/hosts . Visible in the [API docs](https://docs.datadoghq.com/api/latest/hosts/?site=us3), you can change the site on top right corner.

Reply:
Ah, that's very helpful to know. Thanks!